File: ECHO.R
ReadStdin <- function(what = integer()) {
  ## Lire un vecteur (par défaut d'entiers) depuis l'entrée standard.
  ## L'argument what permet de choisir le type de données. 
  ## Par exemple, ReadStdin(what = character()) lit un vecteur de chaînes de caractères.
  ##
  ## Capturer le flux d'entrée standard.
  stream <- file("stdin","r") 
  ## Lire le flux jusqu'à la fin dans un vecteur.
  x <- scan(file=stream, what= what, quiet=TRUE) 
  ## Ferme proprement le flux (pour éviter un warning).
  close(stream)
  ## Renvoyer le vecteur lu.
  return(x)
}

WriteStdout <- function(x, collapse = "\n") {
  ## Écrire le vecteur numérique x sur la sortie standard
  ##
  ## Utiliser l'écriture décimale (pas la notation scientifique).
  y <- format(as.numeric(x), scientific = FALSE, trim = TRUE) 
  ## Écrire le vecteur. Par défaut, il y a un nombre par ligne.
  cat(y, sep = collapse) 
  ## Attention, on appelle la fonction cat et non pas print!
}

######################################
## Script principal

## Lecture des données depuis l'entrée standard.
## Stockage des données dans un vecteur.
## Suppression des caractères d'espacement au début et à la fin de chaque chaîne avec strip.
x <- ReadStdin()

## Traitement des données 
## Ici, il n'y a rien à faire pour echo.
y <- x 

## Écriture du résultat sur la sortie standard
WriteStdout(y)

File: ECHO.java
import java.util.Scanner;

/* La classe ne doit pas être publique. 
 * Il est quand même recommandé de nommer le fichier ECHO.java.
 */
class ECHO {
 
  public static void main(String[] args) {
      // Lire un vecteur d'entiers depuis l'entrée standard.
      // La classe scanner permet de lire facilement des types primitifs et des chaînes de caractères.
      final Scanner sc = new Scanner(System.in);
      // Tant qu'il reste un entier à lire
      while(sc.hasNextInt()) {
          // on écrit directement l'entier sur la sortie standard (sans le stocker)
          System.out.println(sc.nextInt());
      }
  }
}
